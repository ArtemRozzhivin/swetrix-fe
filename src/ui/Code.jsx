import React, { memo } from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'

const Code = ({ text, className, language }) => (
  <pre className={cx('whitespace-pre-line w-full rounded-md bg-gray-800', className)}>
    <code className={`language-${language}`}>{text}</code>
  </pre>
)

Code.propTypes = {
  text: PropTypes.oneOfType([
    PropTypes.string, PropTypes.number, PropTypes.node,
  ]).isRequired,
  language: PropTypes.string.isRequired,
  className: PropTypes.string,
}

Code.defaultProps = {
  className: '',
}

export default memo(Code)
