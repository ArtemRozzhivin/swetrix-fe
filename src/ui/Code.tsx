import React, { memo } from 'react'
import PropTypes from 'prop-types'
import cx from 'clsx'

interface ICode {
  text: string | number | React.ReactNode,
  className?: string,
  language: string,
}

const Code = ({ text, className, language }: ICode) => (
  <pre className={cx('w-full rounded-md bg-gray-800 dark:bg-gray-750', className)}>
    <code className={`whitespace-pre-wrap language-${language}`}>{text}</code>
  </pre>
)

Code.propTypes = {
  text: PropTypes.oneOfType([
    PropTypes.string, PropTypes.number, PropTypes.node,
  ]).isRequired,
  language: PropTypes.string.isRequired,
  className: PropTypes.string,
}

Code.defaultProps = {
  className: '',
}

export default memo(Code)
