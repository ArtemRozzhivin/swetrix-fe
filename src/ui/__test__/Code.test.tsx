/* eslint-disable testing-library/prefer-screen-queries */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react'
import { render } from '@testing-library/react'
import Code from '../Code'

describe('Code component', () => {
  const testText = 'const greeting = "Hello, world!";'
  const testLanguage = 'javascript'

  it('renders the correct text and language', () => {
    const { getByText } = render(
      <Code text={testText} language={testLanguage} />,
    )
    const codeElement = getByText(testText)
    expect(codeElement).toBeInTheDocument()
    expect(codeElement).toHaveClass(`language-${testLanguage}`)
  })

  it('applies any additional className passed in as a prop', () => {
    const testClassName = 'test-class'
    const { container } = render(
      <Code text={testText} language={testLanguage} className={testClassName} />,
    )
    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toHaveClass(testClassName)
  })
})
