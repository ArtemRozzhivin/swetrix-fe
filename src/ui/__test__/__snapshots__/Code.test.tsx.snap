// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code component snapshot 1`] = `
<div>
  <pre
    class="w-full rounded-md bg-gray-800 dark:bg-gray-750"
  >
    <code
      class="whitespace-pre-wrap language-javascript"
    >
      const greeting = "Hello, world!";
    </code>
  </pre>
</div>
`;
